web-gateway/app/controllers/AbstractAuctionController.scala

```diff
@@ -1,20 +1,23 @@
 package controllers
 
-import java.time.{Duration, Instant, ZoneId, ZonedDateTime}
 import java.time.format.DateTimeFormatter
 import java.time.temporal.ChronoUnit
-import java.util.{Date, UUID}
+import java.time.{ Duration, Instant, ZoneId, ZonedDateTime }
+import java.util.{ Date, UUID }
 
-import com.example.auction.user.api.{User, UserService}
+import com.example.auction.user.api.{ User, UserService }
 import org.ocpsoft.prettytime.PrettyTime
-import org.ocpsoft.prettytime.impl.{ResourcesTimeFormat, ResourcesTimeUnit}
+import org.ocpsoft.prettytime.impl.{ ResourcesTimeFormat, ResourcesTimeUnit }
 import org.ocpsoft.prettytime.units.JustNow
-import play.api.i18n.{Messages, MessagesApi}
-import play.api.mvc.{Controller, RequestHeader, Result}
+import play.api.i18n.Messages
+import play.api.mvc._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ ExecutionContext, Future }
+
+abstract class AbstractAuctionController(userService: UserService, controllerComponents: ControllerComponents)
```

Comment:
Hum... maybe we should make the [`defaultExecutionContext` implicit](https://github.com/playframework/playframework/blob/10dfaea2f6393dec1e0e40056d93662a3cf54c06/framework/src/play/src/main/scala/play/api/mvc/Controller.scala#L78), so you won't need to inject it all over the place. Can't remember right now if we decide not to make it implicit or if it is just a detail that slipped.

Reply:
From the doc comment, it looks like it was intentional, but it's not clear why.

In any case, we can do it here for sure.